cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpCredentialConnector.java

```diff
@@ -72,16 +66,148 @@ public CloudCredentialStatus create(AuthenticatedContext authenticatedContext) {
+     * a http request through the google's http api which validates the data
+     * set provided from the given GcpContext.
+     *
+     * @param gcpContext the GcpContext instance which credential would be
+     *                   checked.
+     * @throws IOException if something happens while listing the regions,
+     *                     this exception would thrown by the api.
+     */
     private void preCheckOfGooglePermission(GcpContext gcpContext) throws IOException {
+        /* The actual response which comes from the method invocation is ignored, because if
+        the credential is invalid or the request has timed out or something other happens
+        an exception would thrown automatically */
+        gcpContext.getCompute().regions().list(gcpContext.getProjectId()).executeUsingHead();
+    }
+
+    /**
+     * Attempts to get the "error_description" parameter's value from the given
+     * TokenResponseException's content. If there is no "error_description"
+     * parameter, or it has no value or something occurs during the process, an
+     * empty Optional would return.
+     *
+     * @param e The TokenResponseException which content should have a
+     *          "error_description" parameter with a string value.
+     * @return A String Optional with the content of the "error_description"
+     * from the exception, or an empty one.
+     */
+    private Optional<String> getErrDescriptionFromTokenResponse(TokenResponseException e) {
         try {
-            gcpContext.getCompute().regions().list(gcpContext.getProjectId());
-        } catch (NullPointerException ignore) {
-            LOGGER.error(String.format("Google authentication failed for project-id '%s' because of a null value:", gcpContext.getProjectId()), ignore);
+            return getErrDescriptionFromTokenErrorJson(new JsonParser().parse(e.getContent()).getAsJsonObject());
+        } catch (Exception err) {
+            LOGGER.debug("Could not parse TokenResponseException", err);
+            return Optional.empty();
         }
     }
 
-    @Override
-    public CloudCredentialStatus delete(AuthenticatedContext authenticatedContext) {
-        return new CloudCredentialStatus(authenticatedContext.getCloudCredential(), CredentialStatus.DELETED);
+    /**
+     * Attempts to obtain the "error_description" string parameter from the
+     * provided JsonObject as a String Optional. If the provided JsonObject
+     * is null, or there is no "error_description" element in the JsonObject,
+     * than an empty Optional going to return.
+     *
+     * @param tokenResponseExceptionAsJson the actual JsonObject which should
+     *                                     contains a String key called
+     *                                     "error_description".
+     * @return If
+     */
+    private Optional<String> getErrDescriptionFromTokenErrorJson(JsonObject tokenResponseExceptionAsJson) {
```

Comment:
We don't need two implementation with different input types, because they are used once, so would be nice to merge those two methods into one. Personally don't prefer this kind of data conversion.